src/clp_logging/handlers.py

```diff
@@ -792,3 +800,198 @@ def __init__(
         super().__init__(
             open(fpath, mode), enable_compression, timestamp_format, timezone, loglevel_timeout
         )
+
+
+class CLPS3Handler(CLPBaseHandler):
+    """
+    Log is written to stream in CLP IR encoding, and uploaded to s3_bucket
+
+    :param s3_bucket: S3 bucket to upload CLP encoded log messages to
+    """
+
+    def init(self, stream: IO[bytes]) -> None:
+        self.cctx: ZstdCompressor = ZstdCompressor()
+        self.ostream: Union[ZstdCompressionWriter, IO[bytes]] = (
+            self.cctx.stream_writer(self.local_buffer) if self.enable_compression else stream
+        )
+        self.last_timestamp_ms: int = floor(time.time() * 1000)  # convert to ms and truncate
+        self.ostream.write(
+            FourByteEncoder.encode_preamble(
+                self.last_timestamp_ms, self.timestamp_format, self.timezone
+            )
+        )
+
+    def __init__(
+        self,
+        s3_bucket: str,
+        stream: Optional[IO[bytes]] = None,
+        enable_compression: bool = True,
+        timestamp_format: Optional[str] = None,
+        timezone: Optional[str] = None,
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+        part_limit: Optional[int] = None
+    ) -> None:
+        super().__init__()
+        self.closed: bool = False
+        self.enable_compression: bool = enable_compression
+        self.local_buffer: io.BytesIO = io.BytesIO()
```

Comment:
in general it's always a good practice to ensure a buffer is bounded to avoid OOM issues. Can we use BufferedRandom instead?

Reply:
maybe with the buffer_size attribute 
, you don't need to use BufferedRandom. let's resolve the confusion on the buffer_size name first as discussed at its declaration